### PR TITLE
Add sections on Intel ME dependency

### PR DESCRIPTION
**Pull request by plbossart**

### CODE REVIEW

File: getting_started/intel_debug/suggestions.rst
Comment (marc-hb) [inline getting_started/intel_debug/suggestions.rst:41]: As noted by @fredoh9 , maybe mention here that `/dev/mei*/` requires some Kconfig in the first place.

Also mention that firmware authentication itself does NOT require any `/dev/mei*/`, which could be confusing. Calling it a "debug interface" should avoid any confusion even though it's not just a debug interface.

**Comment by plbossart** [comment]
please don't merge for now, the part on /dev/mei is likely to require a significant update.

**Review by lgirdwood — COMMENTED** [review]
This is useful info.

**Comment by plbossart** [comment]
updated (no way to detect ME enablement) and added section on HDaudio amps.

File: getting_started/intel_debug/introduction.rst
Comment (marc-hb) [inline getting_started/intel_debug/introduction.rst:109]: ..., be it signed by an Intel or OEM production key (consumer products) or by a community key (development systems and make boards)

File: getting_started/intel_debug/suggestions.rst
Comment (marc-hb) [inline getting_started/intel_debug/suggestions.rst:52]: ..., more information about the ME in the "firmware binary" section above.

File: getting_started/intel_debug/suggestions.rst
Comment (marc-hb) [inline getting_started/intel_debug/suggestions.rst:50]: without any explicit feedback provided to the user

(more generic)

File: getting_started/intel_debug/introduction.rst
Comment (lgirdwood) [reply getting_started/intel_debug/introduction.rst:109]: Chromebooks also use community key.

**Review by lgirdwood — APPROVED** [review]
@deb-intel are you able to fix up the grammar after merge as you usually do ? @marc-hb has pointed out a few grammar changes., but there maybe others. Need to merge for v1.7

**Comment by plbossart** [comment]
pushed changes suggested by @marc-hb

**Review by marc-hb — APPROVED** [review]

**Review by deb-intel — APPROVED** [review]
Will make grammatical updates in a separate PR.

**Pull request merged**
